src/validator.cc

```diff
@@ -779,42 +768,50 @@ void Validator::CheckMemory(const Location* loc, const Memory* memory) {
 void Validator::CheckDataSegments(const Module* module) {
   for (const ModuleField& field : module->fields) {
     if (auto data_segment_field = dyn_cast<DataSegmentModuleField>(&field)) {
-      DataSegment* data_segment = data_segment_field->data_segment;
+      auto&& data_segment = data_segment_field->data_segment;
```

Comment:
I thought the/our convention was to use auto only when the type is obvious somewhere else in the statement?  (But maybe I'm just mis-remembering).

Reply:
Yeah, probably a good idea. I'll undo these.

Reply:
See #598.